src/resource.rs

```diff
@@ -155,6 +158,15 @@ impl Resource {
         }
     }
 
+    /// Creates a new resource with the given name and requirements.
+    pub fn new_with_reqs(name: ResourceId, resource_reqs: ResourceSet) -> Self {
```

Comment:
How does one pass the `name` then?

Reply:
so the example below would be written
```rust
let mut resource = Resource::new(
        resource_id()).with_reqs(
        ResourceSet::new_from_resources(HashSet::from_iter(vec![
            super::int_types::resource_id(),
            super::float_types::resource_id(),
        ])),
    );

```

Reply:
note that the `with` method consumes `self`

Reply:
Hmm, not sure I prefer this. The `with_reqs` method takes an existing `Resource` and sets its resource set to something (presumably discarding whatever was there). So this is more of a setter than a constructor?

Reply:
sure, but not a mutable one, so you'd only use it at construction time. It's a common way of setting optional data via method chaining (mirroring the iterator method chaining patterns for example)

Reply:
im not too fussed either way, it is a stylistic choice

Reply:
example: https://github.com/CQCL-DEV/hugr/blob/04e8cc6788226247ae9d54535419b0fcd7755dcb/src/types.rs#L92

Reply:
Alright so it does take a mutable object which explains why it works. Still not sure why it's better though, especially since we only want to call it immediately after constructing a thing with a name. Hmm, I guess this is useful if you have a lot of members and want to enable construction with arbitrary subsets of them as non-default.

Reply:
yes sorry my bad - I should have said not a mutable _reference_. 
And yes if you look at that example there's a different `with` method for a different option I think